packages/relay/src/lib/eth.ts

```diff
@@ -778,13 +778,13 @@ export class EthImpl implements Eth {
       input: contractResult.function_parameters,
       maxPriorityFeePerGas: maxPriorityFee,
       maxFeePerGas: maxFee,
-      nonce: contractResult.nonce,
+      nonce: contractResult.nonce === null ? null : EthImpl.numberTo0x(contractResult.nonce),
```

Comment:
nit: Add a nullable param to `EthImpl.numberTo0x()`. 
With default true and use this logic here and a few places.

Reply:
I suggest to create a new static function eg. `EthImpl.nullableNumberTo0x`. If we modify `numberTo0x` we will need to refactor/touch a lot of rpc methods because they rely on the returned type of `numberTo0x` to be a `string`